tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -17,13 +17,96 @@
 
 #include <ROOT/RFieldBase.hxx>
 
+namespace {
+using ROOT::Experimental::RNTupleOpenSpec;
+void EnsureUniqueNTupleNames(const std::vector<RNTupleOpenSpec> &ntuples)
+{
+   auto sortedNTuples = std::vector<RNTupleOpenSpec>(ntuples.begin(), ntuples.end());
+   std::sort(sortedNTuples.begin(), sortedNTuples.end(), [](const RNTupleOpenSpec &left, const RNTupleOpenSpec &right) {
+      return left.fNTupleName < right.fNTupleName;
+   });
+   auto newEnd = std::unique(
+      sortedNTuples.begin(), sortedNTuples.end(),
+      [](const RNTupleOpenSpec &left, const RNTupleOpenSpec &right) { return left.fNTupleName == right.fNTupleName; });
+   if (newEnd != sortedNTuples.end()) {
+      throw ROOT::Experimental::RException(R__FAIL("horizontal joining of RNTuples with the same name is not allowed"));
+   }
```

Comment:
Maybe use a `std::unordered_set<std::string_view>>` to check for duplicates without copying and actually sorting objects?